### PR TITLE
Tnt-42402

### PR DESCRIPTION
**Pull request by shandilya3**
Decouple telemetry handling from ODD service

## Description
Currently telemetry data is only collected and sent for on-device-decisioning requests.  In order to add telemetry to server-side requests we need to do a small refactor to decouple the telemetry handling from ODD handling.

## Motivation and Context

It is a small refactoring to support telemetry for server side & hybrid.

## How Has This Been Tested?
Unit tests was run to make sure it doesn't break  existing flow.
Build the jar & tested on local machine to see if the telemetry data was passed for activity with ODD enabled.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/main/java/com/adobe/target/edge/client/service/NotificationDeliveryService.java
Comment (jasonwaters) [inline src/main/java/com/adobe/target/edge/client/service/NotificationDeliveryService.java:55]: no reason to abbreviate variable names, it's better to have pronounceable names.  `deliveryRequest` is better than `dreq`.  You  may also want to rename `deliveryRequest` in the method signature to `targetDeliveryRequest` to make it work.

Here's a useful summary from [Clean Code](https://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/B08X8ZXT15/ref=sr_1_1?dchild=1&keywords=clean+code&qid=1633382122&sr=8-1) about variable naming.  https://medium.com/coding-skills/clean-code-101-meaningful-names-and-functions-bf450456d90c

File: src/main/java/com/adobe/target/edge/client/service/NotificationDeliveryService.java
Comment (jasonwaters) [inline src/main/java/com/adobe/target/edge/client/service/NotificationDeliveryService.java:60]: You probably just refactored this code, but `notificationRequest` is better than `notifRequest` IMO

**Review by jasonwaters — COMMENTED** [review]
looks good to me, but I defer to @dcottingham for approval.  He's been leading the charge on the telemetry stuff.

File: src/main/java/com/adobe/target/edge/client/service/NotificationDeliveryService.java
Comment (shandilya3) [reply src/main/java/com/adobe/target/edge/client/service/NotificationDeliveryService.java:55]: I didn't modify any code, just moved the method out in it's own class. But lemme fix for better readability.

File: src/main/java/com/adobe/target/edge/client/service/NotificationDeliveryService.java
Comment (dcottingham) [inline src/main/java/com/adobe/target/edge/client/service/NotificationDeliveryService.java:48]: Let's move the `clientConfig.isTelemetryEnabled()` check inside of TelemetryService.createTelemetryEntry so that it's encapsulated within that service.  And then all we care about outside the service is the check for `telemetry.getEntries().isEmpty()`

File: src/main/java/com/adobe/target/edge/client/service/NotificationDeliveryService.java
Comment (dcottingham) [inline src/main/java/com/adobe/target/edge/client/service/NotificationDeliveryService.java:75]: No more isTelemetryEnabled() check outside of the TelemetryService.  Instead just check `noTelemetry` here.

**Review by dcottingham — APPROVED** [review]
Overall looks great!  Just requested a minor change before you merge.

File: src/main/java/com/adobe/target/edge/client/ondevice/OnDeviceDecisioningService.java
Comment (shandilya3) [reply src/main/java/com/adobe/target/edge/client/ondevice/OnDeviceDecisioningService.java:318]: yes

File: src/main/java/com/adobe/target/edge/client/ondevice/OnDeviceDecisioningService.java
Comment (dcottingham) [inline src/main/java/com/adobe/target/edge/client/ondevice/OnDeviceDecisioningService.java:201]: Surprised our Spotless plugin didn't catch this, but please make this `if(telemetryEntry != null)` to be consistent with other places in the code.

**Pull request merged**
